### ISSUE TITLE
Handle events on Task level

### ISSUE THREAD

**Issue by hieu-lam** [aelassas/Wexflow#248]
Hi, currently the events OnSuccess, OnError, OnWarning, OnApproval, OnRejected belong to the workflow level. That means if a task failed so the whole workflow failed. 

Is there a way to handle exception/error/warning on Task level? Something like:
"If task 3 fails, perform task 4. If task 4 fail, perform task 5. Otherwise, perform task 6". 

Thanks in advance.

**Comment by aelassas** [comment]
@hieu-lam Yes, events are on workflow level. If all the tasks of a workflow fail then OnError event is triggered. If only one task fails and other succeed then OnWarning event is triggered. If all the tasks succeed, then OnSuccess event is triggered. For the moment, events are not supported on task level.
